backend/drm/atomic.c

```diff
@@ -9,57 +9,42 @@
 
 struct atomic {
 	drmModeAtomicReq *req;
-	int cursor;
 	bool failed;
 };
 
-static void atomic_begin(struct wlr_drm_crtc *crtc, struct atomic *atom) {
-	if (!crtc->atomic) {
-		crtc->atomic = drmModeAtomicAlloc();
-		if (!crtc->atomic) {
-			wlr_log_errno(WLR_ERROR, "Allocation failed");
-			atom->failed = true;
-			return;
-		}
-	}
-
-	atom->req = crtc->atomic;
-	atom->cursor = drmModeAtomicGetCursor(atom->req);
-	atom->failed = false;
-}
-
-static bool atomic_end(int drm_fd, uint32_t flags, struct atomic *atom) {
-	if (atom->failed) {
-		return false;
-	}
+static void atomic_begin(struct atomic *atom) {
+	memset(atom, 0, sizeof(*atom));
```

Comment:
Because struct atomic has the 2 fields, and you never access them before you set them, this can be removed and you can add `atom->failed = false` after the if right? I could be wrong....

Reply:
Yes, but I prefer to memset the whole struct. It's too easy to add a new field in the struct and forget to initialize it everywhere. That's also the reason why we use calloc instead of malloc throughout the codebase.

Reply:
That makes complete sense.... my bad.